### PR TITLE
Catalog Service using mpRestClient to communicate w/ Inventory

### PR DESCRIPTION
**Pull request by dshimo**
wait I need to test this first

### CODE REVIEW

File: catalog/pom.xml [hunk 1]
```diff
@@ -72,6 +72,12 @@
             <artifactId>javax.ejb-api</artifactId>
             <version>3.2-b02</version>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.rest.client</groupId>
+            <artifactId>microprofile-rest-client-api</artifactId>
+            <version>1.0</version>
+            <scope>provided</scope>
+        </dependency>
```
Comment (Hemankita) [inline catalog/pom.xml:80]: Rest client comes with Microprofile 1.3. There is no need to explicitly include it.

File: catalog/src/main/liberty/config/server.xml [hunk 2]
```diff
@@ -3,13 +3,14 @@
   <featureManager>
       <feature>microprofile-1.2</feature>
       <feature>ejbLite-3.2</feature>
+      <feature>mpRestClient-1.0</feature>
+      <feature>cdi-1.2</feature>
+      <feature>mpConfig-1.1</feature>
```
Comment (Hemankita) [inline catalog/src/main/liberty/config/server.xml:8]: These features are included in microprofile implicitly. Please use microprofile 1.3.

File: catalog/src/main/java/client/InventoryServiceClient.java [hunk 3]
```diff
@@ -1,52 +1,31 @@
...
-	  public String healthCheck(){
-		  Client client = ClientBuilder.newClient();
-		  WebTarget target = client.target(inv_health);
-		  String s = target.request().get(String.class);
-		  return s;
-	  }
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/check")
+    String healthCheck() throws ProcessingException;
```
Comment (yeekangc) [inline catalog/src/main/java/client/InventoryServiceClient.java:29]: We will replace this with MP Health eventually.

Comment (Hemankita) [reply catalog/src/main/java/client/InventoryServiceClient.java:29]: Yeah I am working on it and will replace the same once the mp health implementation is done. For now, I think David is using the default one we already have there. Thank you.

**Pull request merged**